app/forms/support/location_form.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Support
+  class LocationForm < Form
+    MAIN_SITE = 'main site'
+    FIELDS = %i[
+      location_name
+      urn
+      address1
+      address2
+      address3
+      address4
+      postcode
+    ].freeze
+
+    attr_accessor(*FIELDS)
+
+    validates :location_name,
+              :address1,
+              :address3,
+              :postcode,
+              presence: true
+    validates :postcode, postcode: true
+    validates :urn, reference_number_format: { allow_blank: true, minimum: 5, maximum: 6, message: I18n.t('activemodel.errors.models.support/location_form.attributes.urn.format') }
+    validate :site_name_is_unique
+    # should the URN also be unique?
```

Comment:
in which sense?

Reply:
Just a thought, if it stands for Unique Reference Number, should we validate uniqueness?

Reply:
No as we don't control the uniqueness.

Reply:
Coolio